src/compiler/crystal/codegen/warnings.cr

```diff
@@ -46,17 +46,21 @@ module Crystal
   end
 
   class CodeGenVisitor
+    @deprecated_methods_detected = Set(String).new
```

Comment:
Could you use a Set of Annotation nodes instead of method names? So that we could use the annotations on many other places

Reply:
The annotation is stateless. And they are cloned between typed defs. But also this PR is wrong. I will send a push soon that handle better some cases.

I am stress testing it to avoid more pushes and reviews.